### ISSUE TITLE
Alignment Issues with metric component

### ISSUE THREAD

**Issue by prakhargupta1** [mui/mui-toolpad#2469]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Issue:
<img width="1197" alt="Screenshot 2023-08-10 at 6 23 06 PM" src="https://github.com/mui/mui-toolpad/assets/92228082/d5d0e7dc-974f-472f-a81e-2e558ee4c140">

Steps:
1. Add 2 metrics components next to each other on half of the canvas (other half can be blocked by another component, it's chart in the above screenshot)
2. Observe how it overlaps

The issue seems to be with the box of the metric component as it is unable to define proper boundaries.

### Your environment 🌎

Toolpad version 0.1.22 running locally on Mac M2

**Comment by apedroferreira** [comment]
Yeah, I would say this should probably be fixed at the component level, so allowing the Metric component to be more fluid probably even if it won't always look great.
